src/index.ts

```diff
@@ -186,14 +224,17 @@ class FSMerger {
       this._generateMap();
     }
     let { _dirList } = this;
-    let result: string[] = [], errorCount = 0;
+    let result = new Set<string>();
+    let errorCount = 0;
     let fullDirPath = '';
     for (let i=0; i < _dirList.length; i++) {
       let { root } = this.PREFIXINDEXMAP[i];
       fullDirPath = root + '/' + dirPath;
       fullDirPath = fullDirPath.replace(/(\/|\/\/)$/, '');
       if(fs.existsSync(fullDirPath)) {
-        result.push.apply(result, fs.readdirSync(fullDirPath, options));
+        for (let entry of fs.readdirSync(fullDirPath, options)) {
+          result.add(<any>entry);
+        }
```

Comment:
This is kind of gross. The type of `result` here is either `Set<string>` or `Set<Buffer>` but there's just no good way (that I know of without introducing a bunch of code duplication) to get the typechecker to understand that all these invocations to `readdirSync` are done with the same options (and thus have the same output type). So we have to lie about the type of `result` and cast through `any`.

Reply:
Should we change the return statement from this function? 
https://github.com/SparshithNR/fs-merger/blob/3025c5b5b1beead7052d9e46b375add459f595f4/src/index.ts#L245

Shouldn't it be `return [...result]`

Reply:
Yep! Nice catch.